rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -648,6 +659,13 @@ class FunctionItemNode extends AssocItemNode instanceof Function {
   override TypeParam getTypeParam(int i) { result = super.getGenericParamList().getTypeParam(i) }
 
   override Visibility getVisibility() { result = Function.super.getVisibility() }
+
+  override int getNumberOfParameters() {
+    exists(int arr |
+      arr = super.getNumberOfParams() and
+      if super.hasSelfParam() then result = arr + 1 else result = arr
+    )
+  }
```

Comment:
What about adding this to `Callable` instead as it's not tied to path resolution?

Reply:
Ok; calling it `getNumberOfParamsInclSelf`?

Reply:
That sounds very clear 👍